dataland-frontend/src/components/resources/dataRequest/ReviewRequestButtons.vue

```diff
@@ -0,0 +1,259 @@
+      />
+    </OverlayPanel>
+  </div>
+</template>
+
+<script lang="ts">
+import PrimeButton from "primevue/button";
+import { defineComponent, inject, type PropType } from "vue";
+import type Keycloak from "keycloak-js";
+import { getAnsweredDataRequestsForViewPage, patchDataRequestStatus } from "@/utils/RequestUtils";
+import OverlayPanel from "primevue/overlaypanel";
+import { type DataMetaInformation, type DataTypeEnum, type ErrorResponse } from "@clients/backend";
+import SelectReportingPeriodDialog from "@/components/general/SelectReportingPeriodDialog.vue";
+import { ReportingPeriodTableActions, type ReportingPeriodTableEntry } from "@/utils/PremadeDropdownDatasets";
+import { RequestStatus, type StoredDataRequest } from "@clients/communitymanager";
+import PrimeDialog from "primevue/dialog";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import { AxiosError } from "axios";
+
+export default defineComponent({
+  name: "ReviewRequestButtons",
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+    };
+  },
+  components: { PrimeButton, OverlayPanel, SelectReportingPeriodDialog, PrimeDialog },
+  props: {
+    companyId: {
+      type: String,
+      required: true,
+    },
+    framework: {
+      type: String as PropType<DataTypeEnum>,
+      required: true,
+    },
+    mapOfReportingPeriodToActiveDataset: {
+      type: Map,
+      required: true,
+    },
+  },
+  emits: ["isVisible"],
+  computed: {
+    isVisible() {
+      return this.answeredDataRequestsForViewPage.length > 0;
+    },
+  },
+  watch: {
+    isVisible(newStatus: boolean) {
+      this.$emit("isVisible", newStatus);
+    },
+  },
+  data() {
+    return {
+      answeredDataRequestsForViewPage: [] as StoredDataRequest[],
+      dialogIsVisible: false,
+      dialog: "Default\n text.",
+      dialogIsSuccess: false,
+      actionOnClick: ReportingPeriodTableActions.ReopenRequest,
+    };
+  },
+  mounted() {
+    void this.updateAnsweredDataRequestsForViewPage();
+  },
+  methods: {
+    /**
+     * Closes the SuccessModal
+     */
+    closeSuccessModal() {
+      this.dialogIsVisible = false;
+    },
+    /**
+     * Opens the SuccessModal with given dialog
+     * @param dialog desired dialog
+     * @param dialogIsSuccess if false, display error message
+     */
+    openSuccessModal(dialog: string, dialogIsSuccess: boolean = true) {
+      this.dialogIsSuccess = dialogIsSuccess;
+      this.dialog = dialog;
+      this.dialogIsVisible = true;
+    },
+    /**
+     * Makes the api call and updates the list of answered data requests.
+     */
+    async updateAnsweredDataRequestsForViewPage() {
+      this.answeredDataRequestsForViewPage = await getAnsweredDataRequestsForViewPage(
+        this.companyId,
+        this.framework,
+        Array.from((this.mapOfReportingPeriodToActiveDataset as Map<string, DataMetaInformation>).keys()),
```

Comment:
please avoid using "as" if possible.
this conversion could be avoided if you specified mapOfReportingPeriodToActiveDataset as Map<string, DataMetaInformation> in the props

Reply:
yes, will do